### PR TITLE
Module 5. Hooks

### PR DESCRIPTION
**Pull request by boryskrutskykh**

### CODE REVIEW

File: src/hooks/useFetchMovies.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  sortBy: 'title' | 'release_date' = 'release_date',
+  genreFilter?: string,
+) => {
+  const [movies, setMovies] = useState<MovieInfo[]>([]);
+
+  useEffect(() => {
+    const fetchMovies = async () => {
+      try {
+        let url = 'http://localhost:4000/movies';
+        const params: Record<string, string> = {};
```
Comment (Badyanchik) [inline src/hooks/useFetchMovies.ts:17]: Check URLSearchParams
it can be used to create the query params string

File: src/hooks/useFetchMovies.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+export const useFetchMovies = (
+  searchQuery: string,
+  searchBy: 'title' | 'genres' = 'title',
+  sortBy: 'title' | 'release_date' = 'release_date',
+  genreFilter?: string,
+) => {
+  const [movies, setMovies] = useState<MovieInfo[]>([]);
+
+  useEffect(() => {
+    const fetchMovies = async () => {
```
Comment (Badyanchik) [inline src/hooks/useFetchMovies.ts:14]: Please, check what race conditions are, AbortController, and how to use AbortController in order to deal with race conditions in React.
For example: https://www.youtube.com/watch?v=VdAlFWBUGV4

File: src/hooks/useFetchMovies.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { MovieInfo } from '../components/movie-tile/movie-tyle.types';
+
+export const useFetchMovies = (
```
Comment (Badyanchik) [inline src/hooks/useFetchMovies.ts:5]: BTW, that's great that you created separate hook

File: src/hooks/useFetchMovies.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        let url = 'http://localhost:4000/movies';
+        const params: Record<string, string> = {};
+        if (searchQuery) {
+          params.search = encodeURIComponent(searchQuery);
+          params.searchBy = searchBy;
+        }
+        if (genreFilter) {
+          params.filter = genreFilter;
+        }
+        params.sortBy = sortBy;
```
Comment (Badyanchik) [inline src/hooks/useFetchMovies.ts:25]: I tried to sort movies by title locally... looks like it doesn't work. I guess you have to add &sortOrder=asc to the query string.

Comment (boryskrutskykh) [reply src/hooks/useFetchMovies.ts:25]: Updated.

**Pull request merged**